mock_transport/README.md

```diff
@@ -0,0 +1,47 @@
+# Mock testing framework
+
+## How to use
+
+### Writing examples or test cases
+
+If you want to write a test case or an example that supports the mock testing framework all you have to do is to call the `start_transaction` function before interacting with Azure. The function expects a transaction name that must be unique. For example:
+
+```rust
+azure_core::mock_transport::start_transaction("create_database_and_collection");
+```
+
+This call initializes the mock framework and it's all you need to add to your code (*note*: this call resolves in a noop if you do not enable the mock testing framework).
+
+#### Limitations
+
+* The mock testing framework supports async and multithreading but the call order **must** be deterministic. Also, for this same reason, tests cannot be run concurrently (with `cargo test -- --test-threads=1`).
```

Comment:
I've been simplistic - and misleading - in the description of the limitation. In reality we cannot have two concurrent tests sharing the same environmental variable space. That's because the "request number" counter is stored in an environmental variable that, for obvious reasons, is hardcoded. 

So if test A performs request 1,2,3 it goes like this:

1. Init transaction -> set counter to 1.
2. Read request/response 1 and increment the counter to 2.
3. Read request/response 2 and increment the counter to 3.
4. Read request/response 3 and increment the counter to 4.

Now since te counter is a shared environmental variable the "increment the counter" operation would affect any other test running in the same environment space.

It's not a big deal though, the limitation can be simply addressed by not sharing the environment variables ✌️ .